src/dbscan.jl

```diff
@@ -132,15 +132,15 @@ points = randn(3, 10000)
 clusters = dbscan(points, 0.05, min_neighbors = 3, min_cluster_size = 20) # clusters with less than 20 points will be discarded
 ```
 """
-function dbscan{T <: AbstractFloat, N}(points::Array{T, N}, radius::Real; leafsize::Int = 20, kwargs ...)
+function dbscan{T <: Real}(points::AbstractMatrix{T}, radius::T; leafsize::Int = 20, kwargs ...)
     kdtree = KDTree(points; leafsize=leafsize)
     return _dbscan(kdtree, points, radius; kwargs ...)
 end
 
 
 """ An implementation of DBSCAN algorithm that keeps track of both the core and boundary points """
-function _dbscan{T<:AbstractFloat}(kdtree::KDTree, points::Matrix{T}, radius::T;
-                                   min_neighbors::Int = 1, min_cluster_size::Int = 1)
+function _dbscan{T<:Real}(kdtree::KDTree, points::Matrix{T}, radius::T;

```

Comment:
`radius::Real`

We can also use this opportunity and improve the signature:
* remove `T` completely (it's not used in the body)
* set `points` to `::AbstractMatrix` to match `dbscan()`
* it might be also good to make `min_neighbors` and `min_cluster_size` positional since this method is internal and avoiding keyargs should improve type inference, but that's out of the scope of fixing #84